src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -29,9 +32,16 @@
     @Autowired
     private BookRepository bookRepository;
 
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
```

Comment:
Instead of using it here and in `CustomAuthenticationProvider` you could use it directly in the `setPassword` method from `User`. That way, no matters which class use the setPassword method, you know that will be encoded :)